packages/cli/src/credentials/oauth2Credential.api.ts

```diff
@@ -162,6 +163,18 @@ oauth2CredentialController.get(
 			returnUri = `${get(oauthCredentials, 'authUrl', '') as string}?${percentEncoded}`;
 		}
 
+		if (oauthCredentials.grantType === 'pkce') {
+			let { code_verifier, code_challenge } = pkceChallenge();
+			// Twitter OAuth2 doesn't allow ~ in the code_verifier
+			if (code_verifier.includes('~')) {
+				do {
+					({ code_verifier, code_challenge } = pkceChallenge());
+				} while (code_verifier.includes('~'));
+			}
+			returnUri += `&code_challenge=${code_challenge}&code_challenge_method=S256`;
+			codeVerifier = code_verifier;
```

Comment:
instead of storing this in a local variable, we should store it on the db credential, like we do for `csrfSecret`. 
Otherwise parallel authorization flows could mess with each other.